### ISSUE TITLE
#include "config.h" in public header harmful

### ISSUE THREAD

**Issue by reneengelhard** [Numbertext/libnumbertext#3]
Soros.hxx:

```
/* Soros interpreter (see numbertext.org)
 * 2018 (c) László Németh
 * License: LGPL/BSD dual license */

#ifndef SOROS_HXX_
#define SOROS_HXX_

#include <iostream>
#include <iterator>
#include <string>

#ifndef _MSC_VER
#include "config.h"
#endif

#ifdef HAVE_BOOST_REGEX_HPP
  #include <boost/regex.hpp>
#else
  #include <regex>
#endif
```

is bad. That of course fails when building LibreOffice against a system-libnumbertext because the following happens (Soros.hxx is included by Numbertext.hxx):

```
In file included from /usr/include/libnumbertext/Numbertext.hxx:8:0,
                 from /data/rene/git/LibreOffice/master/lingucomponent/source/nu
mbertext/numbertext.cxx:41:
/usr/include/libnumbertext/Soros.hxx:13:10: fatal error: config.h: Datei oder Ve
rzeichnis nicht gefunden
 #include "config.h"
          ^~~~~~~~~~
compilation terminated.
```

(No such file or directory)

Regards,

Rene

**Comment by laszlonemeth** [comment]
Rene, many thanks for the feedback and your quick fix. I will fix it in the upstream code, too.

(Unfortunately, I haven't solved the system installation of the language data, yet, as you likely noticed.)

**Comment by reneengelhard** [comment]
> (Moreover, Unfortunately, I haven't solved the system installation of the language data, yet, as you likely noticed.)

no, that actually works for me. At least they end up properly in /usr/share/libnumbertext :)

**Comment by laszlonemeth** [comment]
Fixed in commit 44dcd84fe31f6aa003a005cc3684d5cc036ae7c4

Rene, I've checked the system installation of the used numbertext-version.h successfully, so I hope, you can remove the Debian patch for config.h removing. Thanks for your bug report!
